### PR TITLE
fix bugs #435 #436

### LINKED ISSUE #435: 'jpeglib.h' file not found on OSX
```
> cmake ../src
-- OpenMP NOT found
-- Using installed EIGEN3 3.3.4
-- Using installed GLEW 2.1.0
-- Using installed GLFW 3.2.1
-- Using installed LIBJPEG 12:0:3
-- Using installed JSONCPP 1.8.4
-- Using installed LIBPNG 1.6.34
-- Building PYBIND11 from source
-- pybind11 v2.2.1
-- Building TINYFILEDIALOGS from source
-- Build experimental projects
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/qianyizh/Desktop/Open3D/build
```
```
> make -j
Scanning dependencies of target PythonHelper
Scanning dependencies of target PythonTest
Scanning dependencies of target tinyfiledialogs
Scanning dependencies of target PythonTutorial
Scanning dependencies of target EncodeShader
Scanning dependencies of target Core
Scanning dependencies of target IO
[  0%] Building CXX object Tools/CMakeFiles/EncodeShader.dir/EncodeShader.cpp.o
[  1%] Building C object External/tinyfiledialogs/CMakeFiles/tinyfiledialogs.dir/tinyfiledialogs.c.o
[  1%] Built target PythonHelper
[  1%] Built target PythonTutorial
[  3%] Building CXX object IO/CMakeFiles/IO.dir/FileFormat/FileBIN.cpp.o
[  3%] Building CXX object IO/CMakeFiles/IO.dir/ClassIO/PointCloudIO.cpp.o
[  3%] Building CXX object IO/CMakeFiles/IO.dir/ClassIO/IJsonConvertibleIO.cpp.o
[  4%] Building CXX object IO/CMakeFiles/IO.dir/ClassIO/PinholeCameraTrajectoryIO.cpp.o
[  4%] Building CXX object IO/CMakeFiles/IO.dir/ClassIO/PoseGraphIO.cpp.o
[  4%] Building CXX object IO/CMakeFiles/IO.dir/FileFormat/FilePLY.cpp.o
[  6%] Building CXX object IO/CMakeFiles/IO.dir/FileFormat/FileJPG.cpp.o
[  7%] Building CXX object IO/CMakeFiles/IO.dir/ClassIO/TriangleMeshIO.cpp.o
[  7%] Building CXX object IO/CMakeFiles/IO.dir/ClassIO/FeatureIO.cpp.o
[  9%] Building CXX object IO/CMakeFiles/IO.dir/FileFormat/FileXYZRGB.cpp.o
[  9%] Building CXX object IO/CMakeFiles/IO.dir/FileFormat/FileJSON.cpp.o
[  9%] Building CXX object IO/CMakeFiles/IO.dir/FileFormat/FileXYZN.cpp.o
[ 10%] Building CXX object IO/CMakeFiles/IO.dir/FileFormat/FilePCD.cpp.o
[ 10%] Building CXX object IO/CMakeFiles/IO.dir/FileFormat/FilePNG.cpp.o
[ 11%] Building CXX object IO/CMakeFiles/IO.dir/FileFormat/FileXYZ.cpp.o
[ 11%] Building CXX object IO/CMakeFiles/IO.dir/FileFormat/FilePTS.cpp.o
[ 11%] Building CXX object IO/CMakeFiles/IO.dir/FileFormat/FileLOG.cpp.o
[ 12%] Building CXX object IO/CMakeFiles/IO.dir/ClassIO/ImageIO.cpp.o
[ 12%] Building C object IO/CMakeFiles/IO.dir/__/External/rply/rply.c.o
[ 13%] Building C object IO/CMakeFiles/IO.dir/__/External/liblzf/lzf_c.c.o
[ 13%] Building C object IO/CMakeFiles/IO.dir/__/External/liblzf/lzf_d.c.o
[ 13%] Built target PythonTest
In file included from /Users/qianyizh/Desktop/Open3D/src/External/liblzf/lzf_d.c:42:
/Users/qianyizh/Desktop/Open3D/src/External/liblzf/lzfP.h:168:6: warning: macro expansion producing 'defined' has undefined behavior [-Wexpansion-to-defined]
#if !STRICT_ALIGN
     ^
/Users/qianyizh/Desktop/Open3D/src/External/liblzf/lzfP.h:82:25: note: expanded from macro 'STRICT_ALIGN'
# define STRICT_ALIGN !(defined(__i386) || defined (__amd64))
                        ^
/Users/qianyizh/Desktop/Open3D/src/External/liblzf/lzfP.h:168:6: warning: macro expansion producing 'defined' has undefined behavior [-Wexpansion-to-defined]
/Users/qianyizh/Desktop/Open3D/src/External/liblzf/lzfP.h:82:44: note: expanded from macro 'STRICT_ALIGN'
# define STRICT_ALIGN !(defined(__i386) || defined (__amd64))
                                           ^
In file included from /Users/qianyizh/Desktop/Open3D/src/External/liblzf/lzf_c.c:42:
/Users/qianyizh/Desktop/Open3D/src/External/liblzf/lzfP.h:168:6: warning: macro expansion producing 'defined' has undefined behavior [-Wexpansion-to-defined]
#if !STRICT_ALIGN
     ^
/Users/qianyizh/Desktop/Open3D/src/External/liblzf/lzfP.h:82:25: note: expanded from macro 'STRICT_ALIGN'
# define STRICT_ALIGN !(defined(__i386) || defined (__amd64))
                        ^
/Users/qianyizh/Desktop/Open3D/src/External/liblzf/lzfP.h:168:6: warning: macro expansion producing 'defined' has undefined behavior [-Wexpansion-to-defined]
/Users/qianyizh/Desktop/Open3D/src/External/liblzf/lzfP.h:82:44: note: expanded from macro 'STRICT_ALIGN'
# define STRICT_ALIGN !(defined(__i386) || defined (__amd64))
                                           ^
/Users/qianyizh/Desktop/Open3D/src/External/liblzf/lzf_c.c:160:5: warning: macro expansion producing 'defined' has undefined behavior [-Wexpansion-to-defined]
#if STRICT_ALIGN
    ^
/Users/qianyizh/Desktop/Open3D/src/External/liblzf/lzfP.h:82:25: note: expanded from macro 'STRICT_ALIGN'
# define STRICT_ALIGN !(defined(__i386) || defined (__amd64))
                        ^
/Users/qianyizh/Desktop/Open3D/src/External/liblzf/lzf_c.c:160:5: warning: macro expansion producing 'defined' has undefined behavior [-Wexpansion-to-defined]
/Users/qianyizh/Desktop/Open3D/src/External/liblzf/lzfP.h:82:44: note: expanded from macro 'STRICT_ALIGN'
# define STRICT_ALIGN !(defined(__i386) || defined (__amd64))
                                           ^
2 warnings generated.
4 warnings generated.
[ 14%] Linking CXX executable ../bin/EncodeShader
[ 14%] Built target EncodeShader
[ 14%] Building CXX object Core/CMakeFiles/Core.dir/Registration/GlobalOptimization.cpp.o
[ 14%] Building CXX object Core/CMakeFiles/Core.dir/Registration/FastGlobalRegistration.cpp.o
[ 15%] Building CXX object Core/CMakeFiles/Core.dir/Registration/Registration.cpp.o
[ 16%] Building CXX object Core/CMakeFiles/Core.dir/Registration/PoseGraphFactory.cpp.o
[ 17%] Building CXX object Core/CMakeFiles/Core.dir/Registration/ColoredICP.cpp.o
[ 17%] Building CXX object Core/CMakeFiles/Core.dir/Registration/PoseGraph.cpp.o
[ 18%] Building CXX object Core/CMakeFiles/Core.dir/Registration/TransformationEstimation.cpp.o
[ 18%] Building CXX object Core/CMakeFiles/Core.dir/Registration/CorrespondenceChecker.cpp.o
Scanning dependencies of target ShaderFileTarget
[ 19%] Building CXX object Core/CMakeFiles/Core.dir/Registration/Feature.cpp.o
[ 19%] Generating /Users/qianyizh/Desktop/Open3D/src/Visualization/Shader/Shader.h
[ 19%] Building CXX object Core/CMakeFiles/Core.dir/Integration/UniformTSDFVolume.cpp.o
[ 20%] Building CXX object Core/CMakeFiles/Core.dir/Integration/ScalableTSDFVolume.cpp.o
[ 21%] Building CXX object Core/CMakeFiles/Core.dir/Camera/PinholeCameraTrajectory.cpp.o
[ 21%] Building CXX object Core/CMakeFiles/Core.dir/Camera/PinholeCameraTrajectoryFactory.cpp.o
[ 22%] Building CXX object Core/CMakeFiles/Core.dir/Camera/PinholeCameraIntrinsic.cpp.o
[ 22%] Building CXX object Core/CMakeFiles/Core.dir/Geometry/LineSetFactory.cpp.o
[ 23%] Building CXX object Core/CMakeFiles/Core.dir/Geometry/PointCloud.cpp.o
[ 24%] Building CXX object Core/CMakeFiles/Core.dir/Geometry/TriangleMesh.cpp.o
[ 24%] Building CXX object Core/CMakeFiles/Core.dir/Geometry/Image.cpp.o
[ 25%] Building CXX object Core/CMakeFiles/Core.dir/Geometry/KDTreeFlann.cpp.o
[ 25%] Building CXX object Core/CMakeFiles/Core.dir/Geometry/LineSet.cpp.o
[ 26%] Building CXX object Core/CMakeFiles/Core.dir/Geometry/TriangleMeshFactory.cpp.o
[ 26%] Built target ShaderFileTarget
[ 26%] Building CXX object Core/CMakeFiles/Core.dir/Geometry/PointCloudFactory.cpp.o
[ 27%] Building CXX object Core/CMakeFiles/Core.dir/Geometry/ImageFactory.cpp.o
[ 28%] Building CXX object Core/CMakeFiles/Core.dir/Geometry/RGBDImage.cpp.o
[ 28%] Building CXX object Core/CMakeFiles/Core.dir/Geometry/RGBDImageFactory.cpp.o
[ 29%] Building CXX object Core/CMakeFiles/Core.dir/Geometry/EstimateNormals.cpp.o
Scanning dependencies of target Visualization
[ 29%] Building CXX object Core/CMakeFiles/Core.dir/Geometry/DownSample.cpp.o
[ 30%] Building CXX object Core/CMakeFiles/Core.dir/ColorMap/ColorMapOptimization.cpp.o
[ 31%] Building CXX object Core/CMakeFiles/Core.dir/Odometry/Odometry.cpp.o
[ 31%] Building CXX object Core/CMakeFiles/Core.dir/Odometry/RGBDOdometryJacobian.cpp.o
[ 32%] Building CXX object Core/CMakeFiles/Core.dir/Utility/Console.cpp.o
[ 32%] Building CXX object Core/CMakeFiles/Core.dir/Utility/Timer.cpp.o
[ 33%] Building CXX object Core/CMakeFiles/Core.dir/Utility/FileSystem.cpp.o
[ 33%] Building CXX object Core/CMakeFiles/Core.dir/Utility/Helper.cpp.o
[ 34%] Building CXX object Core/CMakeFiles/Core.dir/Utility/Eigen.cpp.o
[ 35%] Building CXX object Core/CMakeFiles/Core.dir/Utility/IJsonConvertible.cpp.o
[ 36%] Linking C static library ../../lib/libtinyfiledialogs.a
[ 36%] Built target tinyfiledialogs
[ 36%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Shader/PhongShader.cpp.o
[ 37%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Shader/GeometryRenderer.cpp.o
[ 37%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Shader/SimpleShader.cpp.o
[ 38%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Shader/Simple2DShader.cpp.o
[ 38%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Shader/NormalShader.cpp.o
[ 39%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Shader/ImageMaskShader.cpp.o
[ 40%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Shader/ImageShader.cpp.o
[ 41%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Shader/ShaderWrapper.cpp.o
[ 41%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Shader/SimpleBlackShader.cpp.o
[ 41%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Shader/PickingShader.cpp.o
[ 42%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Visualizer/Visualizer.cpp.o
[ 43%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Visualizer/ViewControlWithEditing.cpp.o
[ 43%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Visualizer/ViewParameters.cpp.o
[ 44%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Visualizer/RenderOption.cpp.o
[ 44%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Visualizer/ViewControlWithCustomAnimation.cpp.o
[ 45%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Visualizer/VisualizerWithEditing.cpp.o
[ 45%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Visualizer/VisualizerWithKeyCallback.cpp.o
[ 45%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Visualizer/RenderOptionWithEditing.cpp.o
[ 47%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Visualizer/ViewControl.cpp.o
[ 47%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Visualizer/VisualizerWithCustomAnimation.cpp.o
[ 48%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Visualizer/VisualizerCallback.cpp.o
[ 49%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Visualizer/VisualizerRender.cpp.o
[ 49%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Utility/ColorMap.cpp.o
[ 51%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Utility/GLHelper.cpp.o
[ 51%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Utility/BoundingBox.cpp.o
[ 51%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Visualizer/ViewTrajectory.cpp.o
[ 51%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Utility/PointCloudPicker.cpp.o
[ 53%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Utility/SelectionPolygon.cpp.o
[ 53%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Utility/DrawGeometry.cpp.o
[ 53%] Building CXX object Visualization/CMakeFiles/Visualization.dir/Utility/SelectionPolygonVolume.cpp.o
/Users/qianyizh/Desktop/Open3D/src/IO/FileFormat/FileJPG.cpp:29:10: fatal error: 'jpeglib.h' file not found
#include <jpeglib.h>
         ^~~~~~~~~~~
1 error generated.
make[2]: *** [IO/CMakeFiles/IO.dir/FileFormat/FileJPG.cpp.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [IO/CMakeFiles/IO.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
[ 53%] Built target Visualization
[ 53%] Built target Core
make: *** [all] Error 2
```

All dependency libraries are installed via brew install.

### PR DESCRIPTION
**Pull request by qianyizh**
This PR fixes #435 #436 and brings back #394

### CODE REVIEW

**Comment by qianyizh** [comment]
I also removed gcc@7 from install script. As I think it takes too much time to install and it is not a necessary component.

File: src/External/CMakeLists.txt [hunk 1]
```diff
@@ -173,13 +173,22 @@ list(APPEND EXTERNAL_INCLUDE_DIRS
      ${GLEW_INCLUDE_DIRS}
      ${GLFW_INCLUDE_DIRS}
      ${JPEG_INCLUDE_DIR}
```
Comment (syncle) [inline src/External/CMakeLists.txt]: Do we need to remove ``JPEG_INCLUDE_DIR``?

File: src/External/CMakeLists.txt [hunk 2]
```diff
@@ -190,4 +199,5 @@ list(APPEND EXTERNAL_LIBRARIES
      ${tinyfiledialogs_LIBRARIES})
 
 set(EXTERNAL_INCLUDE_DIRS ${EXTERNAL_INCLUDE_DIRS} PARENT_SCOPE)
+set(EXTERNAL_LIBRARY_DIRS ${EXTERNAL_LIBRARY_DIRS} PARENT_SCOPE)
 set(EXTERNAL_LIBRARIES    ${EXTERNAL_LIBRARIES}    PARENT_SCOPE)
```
Comment (syncle) [inline src/External/CMakeLists.txt]: Do we need to maintain both ``EXTERNAL_LIBRARY_DIRS`` and ``EXTERNAL_LIBRARIES``?

File: src/External/CMakeLists.txt (hunk 1 above)
Comment (qianyizh) [reply src/External/CMakeLists.txt]: My hypothesis is that for some versions of libjpeg it returns JPEG_INCLUDE_DIR for other versions it returns JPEG_INCLUDE_DIRS.

File: src/External/CMakeLists.txt (hunk 2 above)
Comment (qianyizh) [reply src/External/CMakeLists.txt]: I think we need. ``EXTERNAL_LIBRARIES`` are the raw library names. ``EXTERNAL_LIBRARY_DIRS`` are the search directories. Some compilers are not smart enough to search the OS to find the libraries (Visual Studio, Xcode, etc), so both have to be specified in CMakeLists.

**Pull request merged**